Resources/public/js/form/errors.js

```diff
@@ -0,0 +1,223 @@
+(function ($) {
+    'use strict';
+
+    //
+    // Form errors.
+    // ----------------------------------------------------------------------------------------------------------------
+
+    /**
+     * @typedef {{
+     *   pane: jQuery,
+     *   trigger: jQuery,
+     *   hasError: boolean
+     * }} TabType
+     */
+
+     var forEach = Array.prototype.forEach;
+     var map = Array.prototype.map;
+
+    /**
+     * Returns true if we shouldn't set the aria-invalid attribute on an input.
+     * (e.g. disabled, button, etc...)
+     *
+     * @param {HTMLElement} input
+     * @returns {boolean}
+     */
+    function cannotBeInvalid (input) {
+        return !!(
+            input.disabled
+            || input.readonly
+            || ['hidden', 'button', 'submit', 'reset', 'image'].indexOf(input.type) > -1
+        );
+    }
+
+    /**
+     * Returns whether element contains descendants latching selector.
+     *
+     * @param {string} selector
+     * @param {jQuery|HTMLElement} element
+     * @returns {boolean}
+     */
+    function containsSelector (selector, element) {
+        return !!$(element).find(selector).length;
+    }
+    var containsClientSideError = containsSelector.bind(null, ':invalid');
+    var containsServerSideError = containsSelector.bind(null, '.sonata-ba-field-error');
+
+    /**
+     * Returns an array of TabType objects for the given form.
+     *
+     * @param {jQuery} $form
+     * @param {Function} hasError
+     * @returns {TabType[]}
+     */
+    function getFormTabs ($form, hasError) {
+        return map.call($form.find('.tab-pane'), function (tab) {
+            var $trigger = $form.find('[data-toggle="tab"][href="#' + tab.id + '"]');
+            return {
+                pane: $(tab),
+                trigger: $trigger,
+                hasError: hasError(tab)
+            };
+        });
+    }
+
+    /**
+     * Returns a TabType objects contains an error.
+     *
+     * @param {TabType} tab
+     * @returns {boolean}
+     */
+    function tabHasError (tab) {
+        return !!tab.hasError;
+    }
+
+    /**
+     * Filters out tabs that do not have errors.
+     *
+     * @param {TabType[]} tabs
+     * @returns {TabType[]}
+     */
+    function keepInvalidTabs (tabs) {
+        return tabs.filter(tabHasError);
+    }
+
+    /**
+     * Mark tabs containing errors.
+     *
+     * @param {TabType[]} tabs
+     * @returns {TabType[]}
+     */
+    function markInvalidTabs (tabs) {
+        tabs.forEach(function (tab) {
+            if (tabHasError(tab)) {
+                tab.trigger.parent().addClass('has-error');
+            } else {
+                tab.trigger.parent().removeClass('has-error');
```

Comment:
Could be written in a more concies manner with `toggleClass()` : 

`tab.trigger.parent().toggleClass('has-error', tabHasError(tab));`